src/panels/config/integrations/integration-panels/zwave_js/dialog-zwave_js-add-node.ts

```diff
@@ -0,0 +1,302 @@
+                ></ha-svg-icon>
+                <div class="status">
+                  <p>
+                    ${this.hass.localize(
+                      "ui.panel.config.zwave_js.add_node.inclusion_failed"
+                    )}
+                  </p>
+                </div>
+              </div>
+              <mwc-button slot="primaryAction" @click=${this._close}>
+                ${this.hass.localize("ui.panel.config.zwave_js.common.close")}
+              </mwc-button>
+            `
+          : ``}
+        ${this._status === "finished"
+          ? html`
+              <div class="flex-container">
+                <ha-svg-icon
+                  .path=${mdiCheckCircle}
+                  class="success"
+                ></ha-svg-icon>
+                <div class="status">
+                  <p>
+                    ${this.hass.localize(
+                      "ui.panel.config.zwave_js.add_node.inclusion_finished"
+                    )}
+                  </p>
+                  <a href="${`/config/devices/device/${this._device!.id}`}">
+                    <mwc-button
+                      >${this.hass.localize(
+                        "ui.panel.config.zwave_js.add_node.view_device"
+                      )}</mwc-button
+                    >
+                  </a>
+                </div>
+              </div>
+              <mwc-button slot="primaryAction" @click=${this._close}>
+                ${this.hass.localize("ui.panel.config.zwave_js.common.close")}
+              </mwc-button>
+            `
+          : ``}
+      </ha-dialog>
+    `;
+  }
+
+  private async _secureInclusionToggleChanged(ev): Promise<void> {
+    const target = ev.target;
+    this._use_secure_inclusion = target.checked;
+  }
+
+  private _startInclusion(): void {
+    if (!this.hass) {
+      return;
+    }
+    this._subscribed = this.hass.connection.subscribeMessage(
+      (message) => this._handleMessage(message),
+      {
+        type: "zwave_js/add_node",
+        entry_id: this.entry_id,
+        secure: this._use_secure_inclusion,
+      }
+    );
+    this._addNodeTimeoutHandle = window.setTimeout(
+      () => this._unsubscribe(),
+      90000
+    );
+  }
+
+  private _handleMessage(message: any): void {
+    if (message.event === "inclusion started") {
+      this._status = "started";
+    }
+    if (message.event === "inclusion failed") {
+      this._unsubscribe();
+      this._status = "failed";
+    }
+    if (message.event === "inclusion stopped") {
+      if (this._status !== "finished") {
+        this._status = "";
+      }
+      this._unsubscribe();
+    }
+    if (message.event === "device registered") {
+      this._device = message.device;
+      this._status = "finished";
+      this._unsubscribe();
+    }
+  }
+
+  private _unsubscribe(): void {
+    if (this._status === "started") {
+      this.hass.callWS({
+        type: "zwave_js/stop_inclusion",
+        entry_id: this.entry_id,
+      });
```

Comment:
Will this cause an `inclusion stopped` event?

Reply:
If the Z-Wave controller was in inclusion mode it will trigger an `inclusion stopped` event, otherwise no.

Reply:
So we then call unsubscribe twice?

Reply:
Moved the actual unsubscribe part up so we unsubscribe the listener before firing the stop_inclusion/stop_exclusion websocket command